### ISSUE TITLE
Support Jupyter Notebooks

### ISSUE THREAD

**Issue by tylere** [gee-community/GEE-Dev-Docs#6]
Markdown tutorials are nice, but non-interactive. It would be nice to support Jupyter Notebooks, which could be rendered statically (examples: [GitHub](https://github.com/jupyter/jupyter/wiki/A-gallery-of-interesting-Jupyter-Notebooks), [Colab](https://colab.research.google.com/notebooks/basic_features_overview.ipynb)) or opened up in a users's local Jupyter environment (example: [jupyterlab-github](https://github.com/jupyterlab/jupyterlab-github)) or on a hosted Jupyter server (example: [Open in Colab](https://jakevdp.github.io/PythonDataScienceHandbook/02.01-understanding-data-types.html))

Relevant technologies:
- [nbconvert](https://github.com/jupyter/nbconvert)
- [notedown](https://github.com/aaren/notedown)
- [jupyterlab-github](https://github.com/jupyterlab/jupyterlab-github)

**Comment by guy1ziv2** [comment]
Colab is great, and I found it is very good tool for GEE & Python API. The difficulty is that most GEE users are using Javascript. Also, despite some progress - there isn't a really good way to show multi-layer maps, interactively zoom, allow creation of geometries such as region-of-interest or training set etc. The Earth Engine Code Editor is very convenient for newcomers.

**Comment by slattery** [comment]
Yes, I wish jupyter and node were better friends!    I have a hello-world level test that works on observablehq, but there are some accommodations that need to be made to run the js there vs google's own env.

**Comment by guy1ziv2** [comment]
I noticed some observablehq on GEE here https://beta.observablehq.com/collection/@donmccurdy/earth-engine but it doesn't look very clean way for users....

**Comment by tylere** [comment]
Here is an example of a project (TensorFlow) that accepts contributions as Markdown or Jupyter Notebooks:
https://www.tensorflow.org/community/contribute/documentation

**Comment by jdbcode** [comment]
@tylere Thanks for sharing the TensorFlow example of community contributions. It certainly provides some good ideas. It looks like they use the same (or similar) rendering system as the GEE User Guide - do you know what it is? A couple of convenience links for later reference:

- Tutorial example: https://www.tensorflow.org/tutorials/images/hub_with_keras
- Docs repo: https://github.com/tensorflow/docs

**Comment by slattery** [comment]
Correct Guy, my attempt forks @donmccurdy's - and you are right that for
this use case observable doesn't seem to match up perfectly.

Observable has promise for GEE API use that takes place only in the
browser.   It does appear unsuitable for materials that need to have more
cut-and-paste fidelity with what must be a nodejs based exec model in GEE's
IDE.

I want to try runkit.com next to see if that can get us apples-to-apples.
 Has anyone tried it?

**Comment by guy1ziv2** [comment]
It can work, although I did not manage yet to put a visualization map see
https://runkit.com/guy1ziv2/test-gee  . This possibly can be embedded into
tutorials see https://runkit.com/docs/embed and as it has a way to be
created programmatically (see link), I imagine it would be possible to make
a wrapper that puts the authorization and ee.initialize() code around the
script?

Guy Ziv
